src/sdk/namespace_s3.js

```diff
@@ -139,14 +139,21 @@ class NamespaceS3 {
             // If set, part_number is positive integer from 1 to 10000.
             // Usually part number is not provided and then we read a small "inline" range
             // to reduce the double latency for small objects.
-            if (!params.part_number) {
+            const can_use_get_inline = !params.part_number && !request.Key.endsWith('/');
```

Comment:
also here - missing comment to explain these conditions and why they exist

Reply:
added comment.